File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/control/path/LockedPathPoint.java
package org.firstinspires.ftc.teamcode.control.path;

import org.firstinspires.ftc.teamcode.control.system.Functions;

public class LockedPathPoint extends PathPoint {
    public double h;
    public LockedPathPoint(String signature, double x, double y, double followDistance, double h) {
        super(signature, x, y, followDistance);
        this.h = h;
    }

    public LockedPathPoint(String signature, double x, double y,  double followDistance, double h, Functions.Function func) {
        super(signature, x, y, followDistance, func);
        this.h = h;
    }

    public LockedPathPoint(LockedPathPoint p) {
        this(p.signature, p.x, p.y, p.followDistance, p.h, p.func);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/control/path/LockedPathPoint.kt
package org.firstinspires.ftc.teamcode.control.path

import org.firstinspires.ftc.teamcode.control.system.Functions
import org.firstinspires.ftc.teamcode.util.Angle

open class LockedPathPoint(
    signature: String,
    x: Double,
    y: Double,
    followDistance: Double,
    val h: Angle,
    func: Functions.Function? = null
) : PathPoint(signature, x, y, followDistance, func) {

    override val copy: PathPoint get() = LockedPathPoint(signature, x, y, followDistance, h, func)
    override fun toString(): String {
        return String.format(
            "%s, %.1f, %.1f, %.1f, %.1f, %s",
            signature,
            x,
            y,
            followDistance,
            h,
            func
        )
    }
}
